griptape/config/azure_openai_structure_config.py

```diff
@@ -96,7 +103,15 @@ class AzureOpenAiStructureConfig(StructureConfig):
         kw_only=True,
     )
     vector_store_driver: BaseVectorStoreDriver = field(
-        default=Factory(lambda self: LocalVectorStoreDriver(embedding_driver=self.embedding_driver), takes_self=True),
+        default=Factory(
+            lambda self: self._factory(
+                LocalVectorStoreDriver, "vector_store_driver", embedding_driver=self.embedding_driver
+            ),
+            takes_self=True,
+        ),
         metadata={"serializable": True},
         kw_only=True,
     )
+
+    def __attrs_post_init__(self):
+        self.vector_store_driver = LocalVectorStoreDriver(embedding_driver=self.embedding_driver)
```

Comment:
Can we do `Factory(lambda self: self.embedding_driver)` here?

Reply:
there might have been an issue where the embedding driver was not getting initialized before the Factory method was being called, resulting in the vector store driver having a `DummyEmbeddingDriver`. still trying to reproduce